### DISCUSSION TITLE
Charging a Tesla vehicle: how often should you wake it?

### DISCUSSION THREAD

**Discussion by msmithstubbs** [amberelectric/public-api#151] [Q&A]
Like a few others I'm working on a script that manages charging a Tesla Model 3 to make the most of cheap power from Amber.

One issue I've found is the vehicle is often 'asleep': ie I can't get the current state of charge without waking it. There's an API call to wake it so I'm wondering if anyone has experience with how often to do this? Is every 15 minutes a bad idea? Every hour?

**Comment by rob0101** [comment]
Perhaps not the answer you want, but why wake it ?    Store the charge level, if the car is asleep in 15 minutes the level will not have changed.  It needs to be awake to gain or loose charge.

**Comment by msmithstubbs** [comment]
Thanks for your reply. You make a  good point. My concern was that if I'm polling every 15 minutes I might miss the time when the vehicle is awake, but in practice it's likely to be awake at some point during the day while driving. 

I'll try that: if the vehicle is awake, store the state of charge. If it's asleep I'll assume the stored value is roughly correct.

**Reply by sskumars** [reply]
I am looking forward to implementing something like this to charge my Tesla. I was curious in which library are used out there to interact with the car? I have came across some implementations like https://www.teslaapi.io/ which seems like the request is made via third party website which has security concerns for me. I prefer to host the app on my home server, especially the ones that interact with car. Any suggestions in the right direction would be appreciated.

**Reply by msmithstubbs** [reply]
I've been using the API documented here:

https://tesla-api.timdorr.com

The API is just calls to a server at api.teslamotors.com so you shouldn't need to go through any third party services.  
Once you've got the API tokens you can call the API yourself to start or stop the car charging. They are just HTTP calls so you can use whatever client you like to do this.

I've built a basic script in typescript to run on Cloudflare workers. Once I'm happy with it I'll publish code in case you're interested.

I see this as a short term solution and looking forward to Amber offering official support for smart EV charging.

**Reply by sskumars** [reply]
Thanks for the clarification, I misunderstood teslamotors.com to be thrid party provider. I am also looking for Amber to come out with smart EV charging. Do share your approach when you are happy with it, will be keen to learn more. Cheers.
